File: src/main/java/it/stanislas/kafka/delay/streamjoin/ProducerFactory.java
package it.stanislas.kafka.delay.streamjoin;

import it.stanislas.kafka.delay.model.MessageA;
import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.common.serialization.StringSerializer;

import java.util.Properties;

public class ProducerFactory {
    protected static KafkaProducer<String, String> buildProducer(final String bootstrapServers) {
        final Properties producerConfig = new Properties();
        producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
        producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
        producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
        final KafkaProducer kafkaProducer = new KafkaProducer<String, MessageA>(producerConfig);
        return kafkaProducer;
    }
}

File: src/main/kotlin/it/stanislas/kafka/delay/streamjoin/ProducerFactory.kt
package it.stanislas.kafka.delay.streamjoin

import org.apache.kafka.clients.producer.KafkaProducer
import org.apache.kafka.clients.producer.ProducerConfig
import org.apache.kafka.common.serialization.StringSerializer
import java.util.*

object ProducerFactory {
    fun buildProducer(bootstrapServers: String): KafkaProducer<String, String> {
        val producerConfig = Properties()
        producerConfig[ProducerConfig.BOOTSTRAP_SERVERS_CONFIG] = bootstrapServers
        producerConfig[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] = StringSerializer::class.java.name
        producerConfig[ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG] = StringSerializer::class.java.name
        return KafkaProducer(producerConfig)
    }
}